pandas/core/indexes/datetimes.py

```diff
@@ -286,16 +281,16 @@ def __new__(cls, data=None,
             verify_integrity = True
 
         if data is None:
-            dtarr = DatetimeArray._generate_range(
```

Comment:
> We want the warning after the array creation, in case the array creation raises an exception.

Makes sense, will change.

> I'm not sure whether you want cls(...) or cls._simple_new right now. Probably simple_new

_simple_new.  It's performant and correct regardless of what becomes of the discussed changes in DTA/TDA constructors/validation.